### PR TITLE
Introduce LXC backend

### PR DESCRIPTION
**Pull request by AndrewPashkin**
WIP, I want to introduce tests for that

### CODE REVIEW

**Comment by AndrewPashkin** [comment]
Should I do that, by the way? Since currently backends are not tested.

**Comment by AndrewPashkin** [comment]
@philpep, please, pay attention to "sudo" part of the backend, I'm not sure If I'm using it correctly.

File: testinfra/backend/lxc.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    def __init__(self, container, *args, **kwargs):
+        self.container = container
+        super(LXCBackend, self).__init__(self.container, *args, **kwargs)
+
+    def run(self, command, *args, **kwargs):
+        command = self.get_command(command, *args)
+        template = 'lxc-attach --name %s -- /bin/sh -c %s'
+        if self.sudo:
+            template = 'sudo ' + template
```
Comment (philpep) [inline testinfra/backend/lxc.py]: Indeed you must remove this code, `sudo` is handled in `get_command()`.
If you need sudo to run lxc-attach you must run `sudo testinfra`.

**Comment by philpep** [comment]
I suppose lxc have a way to list hosts you could implement the `get_hosts()` method to handle wildcard and so on using the `fnmatch` module.

For instance:
```
testinfra --connection=lxc # run test on all containers
testinfra --connection=lxc --hosts 'web-*' # run test on all containers names starting with web-
```

**Comment by AndrewPashkin** [comment]
> `testinfra --connection=lxc # run test on all containers`
This can do harmful and undesirable things

**Comment by AndrewPashkin** [comment]
I would suggest to force user to explicitly provide `--hosts='*'`.

**Comment by AndrewPashkin** [comment]
> I suppose lxc have a way to list hosts you could implement the `get_hosts()` method to handle wildcard and so on using the `fnmatch` module.

To implement this, `run_local()` method should be available inside `get_hosts()`, but for that it should be a regular method, to classmethod. What to do with that?

**Comment by AndrewPashkin** [comment]
@philpep Any thoughts on that?

**Comment by philpep** [comment]
>> testinfra --connection=lxc # run test on all containers

> This can do harmful and undesirable things

I agree, I opened #94 for that

**Comment by philpep** [comment]
@AndrewPashkin hi sorry for the delay. Indeed we need to move run_local somewere else.

**Comment by retr0h** [comment]
@philpep would love to see this land.  We recently added libvirt support.

https://github.com/metacloud/molecule/pull/200

**Comment by AndrewPashkin** [comment]
@philpep, should I make decisions?

**Comment by AndrewPashkin** [comment]
@philpep, :thinking:

**Comment by philpep** [comment]
Hi @AndrewPashkin , sorry to have your patches being blocked in this PR without any updates from me, please accept my apologies... I now think that users want a lxc/lxd backend, and it's ok to merge it without support for `get_hosts()`.
 I opened #334 with your commit from #74 cherry-picked on master, I plan to have it in the next release, so I'm closing this PR.
Again, thank you for your work on this and sorry for my very long time to handle this...

**Pull request closed without merging**